cli/runtime/config/nodeutils/replace_nodes.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package nodeutils
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/pkg/errors"
+	"gopkg.in/yaml.v3"
+)
+
+// ReplaceNodes replace nodes in dst as per patchStrategy prior performing merge
+func ReplaceNodes(src, dst *yaml.Node, opts ...PatchStrategyOpts) (bool, error) {
```

Comment:
@mpanchajanya thinking about it more. As implemented the replace logic is done in two phases, first deleting all the nodes marked for 'replacement' by the strategy, then performing the merge from source to destination node, creating the nodes again. 

If the merge logic is not going to be part of this function or replaceNodes that it calls, would it be clearer to call implementation in this file DeleteNodes, deleteNodes?

Reply:
yes makes sense. updated in the multi file PR